Add binary sensor platform to Evohome (#44843)
home-assistant/home-assistant.github.io: source/_integrations/evohome.markdown

### Evohome

Each heating zone is represented as a climate entity that exposes the zone's operating mode, current temperature, and setpoint.

Each zone also provides a button entity to clear any override and return the zone to Evohome's **FollowSchedule** mode.

The system controller is also represented as a climate entity that exposes the system's current operating mode. A controller has neither a current temperature nor a setpoint, but all climate entities in Home Assistant are required to report a temperature, so this value is calculated as the average of all zones.

The controller also provides a button entity to reset the system mode. This returns the system to Evohome's **Auto** mode (or the equivalent, for edge-case systems), and resets all zones and DHW to **FollowSchedule**.

The DHW controller is represented as a water heater entity which will report its current temperature and can be turned on or off. Due to limitations with the vendor's RESTful API, the setpoint is not reported and cannot be changed.

If present, it also provides a button entity to clear any DHW override and return the DHW controller to Evohome's **FollowSchedule** mode.

> Each device in the system—the gateway, the controller, DHW (if present), and each heating zone—also provides a binary sensor that reports active faults. These are diagnostic entities of the `problem` device class: they are _on_ when one or more faults are active, and _off_ otherwise. Each entity exposes a `fault_count` attribute and a `faults` attribute containing the details of any active faults.

Reviewer: Is the "fault_count" attribute still accurate? I could not find it https://github.com/home-assistant/core/pull/168634/changes#diff-39e4228070917a84da10f51ae71ffd87147fdc7a8f8b484f9f4a6fba9555d921R96.

Author: No - sorry for inconveniencing you... The reviewer asked for a change - I neglected to update this PR correctly.

If present, it also provides a **Button** entity to clear any DHW override and return the DHW controller to Evohome's **FollowSchedule** mode.

> Each device in the system—the gateway, the controller, DHW (if present), and each heating zone—also provides a **BinarySensor** entity that reports active faults. These are diagnostic entities of the `problem` device class: they are _on_ when one or more faults are active, and _off_ otherwise. Each entity exposes a `fault_count` attribute and a `faults` attribute containing the details of any active faults.

Reviewer: ```suggestion
Each device in the system—the gateway, the controller, DHW (if present), and each heating zone—also provides a **BinarySensor** entity that reports active faults. These are diagnostic entities of the `problem` device class: they are _on_ when one or more faults are active, and _off_ otherwise. Each entity exposes a `fault_count` attribute and a `faults` attribute containing the details of any active faults. Entity IDs follow the pattern `binary_sensor.<name>_gateway_faults`, `binary_sensor.<name>_system_faults`, `binary_sensor.<name>_dhw_faults`, and `binary_sensor.<zone_name>_zone_faults`.
```
Would it make sense to add a note about naming? Something like this?

Author: I don't think so - the entity names are self-descriptive, and there's the example at the end.

> Each device in the system — the gateway, the controller, DHW (if present), and each heating zone — also provides a **BinarySensor** entity that reports active faults. These are diagnostic entities of the `problem` device class: they are _on_ when one or more faults are active, and _off_ otherwise. Each entity exposes a `fault_count` attribute and a `faults` attribute containing the details of any active faults.

Reviewer: ```suggestion
Each device in the system—the gateway, the controller, DHW (if present), and each heating zone—also provides a **BinarySensor** entity that reports active faults. These are diagnostic entities of the `problem` device class: they are _on_ when one or more faults are active, and _off_ otherwise. Each entity exposes a `fault_count` attribute and a `faults` attribute containing the details of any active faults.
```
We use Merriam Webster's style: no spaces around em dashes. https://www.merriam-webster.com/grammar/em-dash-en-dash-how-to-use

## System modes, zone overrides and inheritance

TCC systems can support up to six distinct operating modes: **Auto**, **AutoWithEco**, **Away**, **DayOff**, **HeatingOff**, and **Custom**. Not all systems support all of these modes.

Zones support three setpoint modes: **FollowSchedule**, **TemporaryOverride**, and **PermanentOverride**, but 'inherit' an operating mode from their system (the actual algorithm for this is a little more complicated than indicated below - please see the vendor's documentation).

For **FollowSchedule**, a zone's `setpoint` (target temperature) is a function of its scheduled target temperature and its inherited mode:

> - **Auto**, setpoints are scheduled temperatures (the default)

Reviewer: ```suggestion
- **Auto** mode follows the scheduled temperature (for example, a scheduled temperature of 20 °C results in a target temperature of 20 °C. **Auto** mode is the default).
```
is it necessary to use the term setpoint? I find target temperature easier to understand. Please feel free to rephrase if my suggestion is inaccurate.

Author: The term used most consistently by the vendor is 'setpoint'.

<img width="486" height="277" alt="image" src="https://github.com/user-attachments/assets/821006a0-fb35-4463-aa65-0c06b337b229" />

Author: So, I'm happy to leave as you suggest, but how about:
```
- **Auto** mode follows the scheduled setpoint (for example, a scheduled setpoint
of 20 °C results in a target temperature of 20 °C. **Auto** mode is the default).
```

Reviewer: I see. Yes, sounds good to me.

Author: I've pushed a change which I think will help - makes a distinction between Evohome setpoints, and Climate entity's target temperature.

> - **AutoWithEco**, setpoints are scheduled temperatures, less 3 °C

Reviewer: ```suggestion
- **AutoWithEco** mode means the target temperature is 3 °C below the scheduled temperature (for example, a scheduled temperature of 20 °C results in a target temperature of 17 °C).
```

If the zone's target temperature is changed then it will either be a **TemporaryOverride** or a **PermanentOverride**, depending. A **TemporaryOverride** will revert to **FollowSchedule** after some specified time. A **PermanentOverride** is a persistent change until some subsequent intervention is made. Zones can be switched between the two override modes without changing the target temperature.

For some system modes all zones will have a setpoint enforced upon them, regardless of their own mode:

> - **Away**, setpoints to 12 °C

Reviewer: ```suggestion
- **Away** mode sets the target temperature to 12 °C.
```

> - **HeatingOff**, setpoints to a minimum, usually 4 °C

Reviewer: ```suggestion
- **HeatingOff** mode sets the target temperature to a predefined minimum, usually 4 °C.
```

{% endraw %}

> All Evohome entities may have faults, which are exposed as **BinarySensor** entities. You can also use them in templates:

Reviewer: ```suggestion
All Evohome entities may have faults, which are exposed as binary sensor entities. You can also use them in templates:
```

{% raw %}

```text
{% if states('binary_sensor.bedroom_faults') == 'on' %}

>   {% if state_attr('binary_sensor.bedroom_faults', 'faults')[0].fault_type == 'TempZoneActuatorLowBattery' %}

Reviewer: ```suggestion
  {% if state_attr('binary_sensor.bedroom_faults', 'faults')[0].fault == 'temp_zone_actuator_low_battery' %}
```
should the key be fault, instead of fault_type?

Author: yes - silly error.

Author: fixed (I ran out of copilot tokens & got lazy)
